packages/app/features/dashboard/beatmaker/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { isSome } from "app/utils";
+import { Text, View } from "app/design-system";
+import ProfileBeatsList from "app/features/dashboard/profile/profileBeatsList";
+import { useBeatmakerParam } from "app/hooks/param/useBeatmakerParam";
+import { UserAvatar } from "app/ui/userAvatar";
+import { PublicUser, useUserCollectionQuery } from "app/api/graphql";
+import { useEffect, useMemo, useState } from "react";
+import { getUser } from "app/api/algolia";
+import useErrorReport from "app/hooks/useErrorReport";
+
+export default function BeatmakerScreen() {
+  const id = useBeatmakerParam();
+  const reportError = useErrorReport();
+  const [displayName, setDisplayName] = useState<string>("");
+  const [avatarUrl, setAvatarUrl] = useState<string | null | undefined>(null);
+  const { data, loading } = useUserCollectionQuery({
+    skip: !id,
+    variables: {
+      userId: id!,
+    },
+  });
+  const entries = data?.entries?.filter(isSome) ?? [];
+
+  const fetchUser = useMemo(
+    () => async () => {
+      if (!id) return;
+
+      let user: PublicUser;
+      try {
+        user = await getUser(id);
+      } catch (e) {
+        reportError(e);
+        return;
+      }
+
+      setAvatarUrl(user.avatarUrl);
+      setDisplayName(user.displayName ?? "");
+    },
+    [id, reportError]
+  );
+
+  useEffect(() => {
+    fetchUser();
+  }, [fetchUser]);
```

Comment:
can we create a custom hook that will receive the fetch function as a param and will return the fetched data, error, and loading? We could reuse it in a few places. The API would be similar to Apollo `useQuery`. We could name it `useFetch` for instance.